src/appshell/view/dockwindow/dockpageview.cpp

```diff
@@ -179,22 +179,36 @@ bool DockPageView::isDockOpen(const QString& dockName) const
 void DockPageView::toggleDock(const QString& dockName)
 {
     DockBase* dock = dockByName(dockName);
+    if (dock == nullptr) {
+        return;
+    }
+
+    bool shouldOpen = true;
+
     DockPanelView* panel = dynamic_cast<DockPanelView*>(dock);
     if (panel) {
-        DockPanelView* destinationPanel = findPanelForTab(panel);
```

Comment:
There was an issue with using findPanelForTab, as it returns the default location for a tab (the first it finds in the allowed panels), not its current location. I've renamed this method findFirstPanelForTab(), and created a new method findCurrentPanelForTab(), which searches for the tab in all valid panels.